### PR TITLE
Reqwest support

### LINKED ISSUE #69: Reqwest support
I've been looking at reqwest and it would allow async multipart #58 and browser support. 

I don't know if it was considered before, since reqwest is built on hyper could be an easy switch.

I'll start a draft PR #70.

edit: browser support not just WASM

### PR DESCRIPTION
**Pull request by SionoiS**
Copy-pasta everywhere without thinking too much. Doesn't compile!

Next step would be to actually think about how to stream into multipart forms.

edit: this PR would resolve #69 #58

### CODE REVIEW

**Comment by SionoiS** [comment]
Ok some progress add_file works!

I'm trying to master the streaming response flow next but it's a big chunk of code.

I kinda understand what need to be done.
1. Send request
2. Read response chunk
3. Deserialize chunk to json
4. Wrap in stream of Result<Json, Error>
And also handle errors but how to do it still escape me...

Staring at your code for couple days should do the trick. :P

**Comment by SionoiS** [comment]
Changed the layout a little.

Also, not sure if keeping hyper make sense, I could remove hyper but keep TLS options since reqwest has them.

Let me know what you think!

Thanks.

**Comment by ferristseng** [comment]
I think this mostly looks good at a first glance, and I'd want to keep hyper just to preserve compatibility. I think I'd prefer handling #58 in a separate PR because I'd like to try to keep the APIs consistent for all of the implementations...which would require modifying the multipart crate first.

File: ipfs-api/Cargo.toml [hunk 1]
```diff
@@ -15,34 +15,39 @@ license                   = "MIT OR Apache-2.0"
 travis-ci                 = { repository = "ferristseng/rust-ipfs-api" }
 
 [features]
-default                   = ["with-hyper-tls", "with-builder"]
+default                   = ["with-actix"]
```
Comment (jcaesar) [inline ipfs-api/Cargo.toml:18]: Odd to add reqwest but change the default to actix? ;)

Comment (SionoiS) [reply ipfs-api/Cargo.toml:18]: good catch! I'll change that.

**Comment by ferristseng** [comment]
Can you see if this will work on top of #72? I'm moving to separate crates to preserve my sanity in maintaining this

**Comment by SionoiS** [comment]
One of the problem is not having reqwest multipart available but otherwise it should work fine.

Don't stress, do the crate split then we can talk about async multipart and then reqwest.

**Pull request closed without merging**